tests/pjit_test.py

```diff
@@ -392,6 +395,51 @@ def testRepeatedOutResources(self):
       with self.assertRaisesRegex(ValueError, error):
         pjit(lambda x: x, in_axis_resources=None, out_axis_resources=spec)(x)
 
+  @with_mesh([('x', 2)])
+  def testInputShardsXMapAxis(self):
+    spec = P('x')
+    f = xmap(pjit(lambda x: x + 2, in_axis_resources=spec, out_axis_resources=None),
+             in_axes=['i', ...], out_axes=['i', ...], axis_resources={'i': 'x'})
+    x = jnp.arange(4).reshape((2, 2))
+    error = (r"pjit input has an axis resources specification of " +
+             spec_regex(spec) + r" which overlaps with resources partitioning "
+             r"named dimensions of that value \(both use mesh axes `x`\)")
+    with self.assertRaisesRegex(JAXTypeError, error):
+      f(x)
+
+  @with_mesh([('x', 2)])
+  def testOutputShardsXMapAxis(self):
+    spec = P('x')
+    f = xmap(pjit(lambda x: x + 2, in_axis_resources=None, out_axis_resources=spec),
+             in_axes=['i', ...], out_axes=['i', ...], axis_resources={'i': 'x'})
+    x = jnp.arange(4).reshape((2, 2))
+    error = (r"pjit output has an axis resources specification of " +
+             spec_regex(spec) + r" which overlaps with resources partitioning "
+             r"named dimensions of that value \(both use mesh axes `x`\)")
+    with self.assertRaisesRegex(JAXTypeError, error):
+      f(x)
+
+  @with_mesh([('x', 2)])
+  def testConstraintShardsXMapAxis(self):
+    spec = P('x')
+    f = xmap(lambda x: with_sharding_constraint(x, axis_resources=spec),
+             in_axes=['i', ...], out_axes=['i', ...], axis_resources={'i': 'x'})
+    x = jnp.arange(4).reshape((2, 2))
+    error = (r"with_sharding_constraint input has an axis resources specification of " +
+             spec_regex(spec) + r" which overlaps with resources partitioning "
+             r"named dimensions of that value \(both use mesh axes `x`\)")
+    with self.assertRaisesRegex(JAXTypeError, error):
+      f(x)
+
```

Comment:
Also test xmap(pjit(sharding_constraint))?

Reply:
I don't think that it would change anything, so I skipped it